alibi_detect/saving/_pytorch/saving.py

```diff
@@ -0,0 +1,134 @@
+import os
+import logging
+from pathlib import Path
+from typing import Any, Callable, Dict, Optional, Tuple, Union
+
+import dill  # dispatch table setting not done here as done in top-level saving.py file
+import torch
+import torch.nn as nn
+
+from alibi_detect.cd.pytorch import UAE, HiddenOutput
+from alibi_detect.models.pytorch import TransformerEmbedding
+from alibi_detect.utils.frameworks import Framework
+
+logger = logging.getLogger(__name__)
+
+
+def save_model_config(model: Callable,
+                      base_path: Path,
+                      input_shape: Optional[tuple],
+                      local_path: Path = Path('.')) -> Tuple[dict, Optional[dict]]:
+    """
+    Save a PyTorch model to a config dictionary. When a model has a text embedding model contained within it,
+    this is extracted and saved separately.
+
+    Parameters
+    ----------
+    model
+        The model to save.
+    base_path
+        Base filepath to save to (the location of the `config.toml` file).
+    input_shape
+        The input dimensions of the model (after the optional embedding has been applied).
+    local_path
+        A local (relative) filepath to append to base_path.
+
+    Returns
+    -------
+    A tuple containing the model and embedding config dicts.
+    """
+    cfg_model = None  # type: Optional[Dict[str, Any]]
+    cfg_embed = None  # type: Optional[Dict[str, Any]]
+    if isinstance(model, UAE):
+        layers = list(model.encoder.children())
+        if isinstance(layers[0], TransformerEmbedding):  # if UAE contains embedding and encoder
+            # embedding
+            embed = layers[0]
+            cfg_embed = save_embedding_config(embed, base_path, local_path.joinpath('embedding'))
+            # preprocessing encoder
+            model = layers[1]
```

Comment:
When `model` is a `UAE`, we unpick the `TransformerEmbedding` and encoder model, and save both separately. This "unpicking" is much more simple than the pytorch equivalent because pytorch models don't need to be built/called before saving (since they already know their dimensions etc). If we replaced `Linear` in `UAE` with `LazyLinear` we would have to do the same as for tensorflow...

Reply:
more simple than `tensorflow` I assume?

Reply:
Oops... yes tensorflow